src/app/modules/round/round.controller.js

```diff
@@ -205,10 +205,21 @@ angular
         }
       }
 
-      // TODO Add docblock
+      /**
+       * @ngdoc method
+       * @name RoundController#cancel
+       * @kind function
+       * @methodOf app.round.controller:cancel
+       * @description
+       * Cancel the last action until the shot is reset
+       */
       function cancel() {
-        // TODO implement
-        alert('to implement');
+        vm.shots[vm.shotIndex] -= vm.number;
```

Comment:
Sembra che questo metodo funzioni solo in un caso. Se sei al terzo tiro e vuoi annullare tutti i tiri, questo metodo te lo permette?